File: pygame/message_screen.py
#!/usr/bin/env python3
import pygame

from settings import *

class MessageScreen:
    # =========================================================
    def __init__(self):
        self.img =  pygame.image.load(RESOURSE_MESSAGE_PATH).convert_alpha()

    # =========================================================
    def draw(self, window):
        cur_screen_width, cur_screen_height = window.get_size()
        pos_x = (cur_screen_width - RESOURCE_MESSAGE_WIDTH) / 2
        pos_y = (cur_screen_height - RESOURCE_MESSAGE_HEIGHT) / 2
        window.blit(self.img, (pos_x, pos_y))

File: sfml/src/MessageScreen.cpp
#include "MessageScreen.h"

#include "settings.h"

MessageScreen :: MessageScreen()
{
    if (!m_texture.loadFromFile(RESOURCE_MESSAGE_PATH))
    {
        throw std::runtime_error(
            "Failed to load message image");
    }

    m_sprite.setTexture(m_texture);
}

void MessageScreen :: draw(sf::RenderWindow& window)
{
    sf::Vector2u windowSize = window.getSize();
    sf::Vector2u textureSize = m_texture.getSize();

    float posX =
        (windowSize.x - textureSize.x) / 2.f;

    float posY =
        (windowSize.y - textureSize.y) / 2.f;

    m_sprite.setPosition(posX, posY);

    window.draw(m_sprite);
}